modules/core/keeper/msg_server.go

```diff
@@ -488,6 +488,11 @@ func (k *Keeper) RecvPacket(goCtx context.Context, msg *channeltypes.MsgRecvPack
 		return nil, errorsmod.Wrap(err, "receive packet verification failed")
 	}
 
+	// performance: return early for the redundant relayer ante handler
+	if (ctx.IsCheckTx() || ctx.IsReCheckTx()) && ctx.ExecMode() != sdk.ExecModeSimulate {
```

Comment:
For backport to v7.5 we should just add a new func which plumbs through the `simulate` bool from ante. We can do similar to the `checkTxUpdateClient` method I added on the ante handler itself. 

There is no ability to check `runTxModeSimulate` on the v0.47 line (before it was changed to "execMode")